src/db.rs

```diff
@@ -81,6 +83,10 @@ pub async fn dataset_select(
 }
 
 /// Insert or Update a Data Product
+// @coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
```

Comment:
@coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
- can we insert a new data product?
- can we update an existing data product?
- do we get rejected if a data products has a reference to non-existent dataset?